specs/agents/tracing-instrumentation-azure-functions.md

```diff
@@ -0,0 +1,111 @@
+actual `faas.trigger.type`. For trigger types that are specifically supported (e.g HTTP) these values
+can vary on trigger-specific specifications apply.
+
+| Field | Value | Description | Source |
+| - | - | - | - |
+| `name` | e.g. `MySampleTrigger` | The transaction name. Use function name if trigger type is `other`. | *generic context* |
+| `type` | e.g. `request`, `messaging` | The transaction type. | Use `request` if trigger type is undefined. |
+| `outcome` | `success`/`failure` | Set to `failure` if a function error can be detected otherwise `success`. | |
+| `result` | `success`/`failure` |Set to `failure` if a function error can be detected, otherwise `success`. | Trigger specific. |
+| `faas.name` | e.g. `MyFunctionApp/MySampleTrigger` | The function app name and the function name, using this format: `<FUNCTION_APP_NAME>/FUNCTION_NAME>`. | *generic context* |
+| `faas.id` | e.g. `/subscriptions/d2ba53be-0815-4...` | The [fully qualified resource ID](https://learn.microsoft.com/en-us/rest/api/resources/resources/get-by-id) of the Azure Function, which has this format: `/subscriptions/<SUBSCRIPTION_GUID>/resourceGroups/<RG>/providers/Microsoft.Web/sites/<FUNCAPP>/functions/<FUNC>` | *generic context*, environment |
+| `faas.trigger.type` | `other` | The trigger type. Use `other` if the trigger type is unknown or cannot be specified. | More concrete triggers are `http`, `pubsub`, `datasource`, `timer` (see specific triggers below). |
+| `faas.execution` | `203621a2-62f...` | The unique invocation id of the function. | *generic context* |
+| `faas.coldstart` | `true`/`false` | A boolean value indicating whether this function invocation was a cold start or not. See the [Deriving cold starts](#deriving-cold-starts) section below. |
+
+### Metadata
+
+Generic cloud metadata fetching SHOULD be disabled when the agent is running in an Azure Functions app. It will not
+yield any useful data and only incur additional cold start overhead due to the non-existent HTTP metadata endpoints.
+
+Agents SHOULD rather:
+
+* Check for the existence of the `FUNCTIONS_WORKER_RUNTIME` environment variable that indicates that the agent is running
+in an Azure Function application.
+* Perform specific cloud metadata detection for Azure Functions (see [Metadata](./metadata.md)).
+
+In addition to those described in [Metadata](./metadata.md), following metadata fields are relevant for Azure Functions:
+
+| Field | Value | Description | Source |
+| - | - | - | - |
+| `service.framework.name` | `Azure Functions` | Constant value for the framework name. | |
+| `service.framework.version` | e.g. `~4` | Version of the Azure Functions runtime. | `FUNCTIONS_EXTENSION_VERSION` |
+| `service.runtime.name`| e.g. `dotnet-isolated` | The language worker runtime (see [here](https://learn.microsoft.com/en-us/azure/azure-functions/functions-app-settings#functions_worker_runtime)). | `FUNCTIONS_WORKER_RUNTIME` |
+
```

Comment:
```suggestion
| `cloud.provider` | `azure` | - | - |
| `cloud.region` | e.g. 'West US 2' | The cloud region. | `REGION_NAME` |
| `cloud.service.name` | 'azure functions' | Constant for the Azure service. See [ECS fields doc](https://www.elastic.co/guide/en/ecs/current/ecs-cloud.html#field-cloud-service-name). | - |
| `cloud.account.id` | e.g. '25b4db88-...' | The cloud id of the function. | Parsed from `WEBSITE_OWNER` if it is defined. `WEBSITE_OWNER_NAME && WEBSITE_OWNER_NAME.split('+', 1)[0]`
```

ECS documents `cloud.service.name` field here: https://www.elastic.co/guide/en/ecs/current/ecs-cloud.html#field-cloud-service-name Our Lambda spec uses "lambda" as a constant for this. Perhaps "functions" (feels too generic) or "azure functions"?

These `metadata.cloud` fields are responsible for the cloud icon next to the service name, e.g.:
<img width="421" alt="Screen Shot 2022-12-15 at 2 21 35 PM" src="https://user-images.githubusercontent.com/46866/207981386-ee0c2804-148e-4f7d-8d35-b303cad9df9a.png">


The `WEBSITE_OWNER_NAME` UUID value matches the value from `az account list`:
```
% az account list
[
  {
    "cloudName": "AzureCloud",
    "homeTenantId": "39f4034f-9fa9-4141-969e-94a6032fad22",
    "id": "25b4db88-9548-4a22-955f-31b826b06078",
    "isDefault": true,
    "managedByTenants": [],
    "name": "[REDACTED]",
    "state": "Enabled",
    ...
```

Reply:
Right, I must have missed this property initially. No intention of not sending it.
As for the name, no strong opinion. So, let's go with `"functions"` (for now).